metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_config.py

```diff
@@ -69,6 +69,9 @@ def __init__(self, **data: Any):
     def get_bigquery_client(self) -> bigquery.Client:
         client_options = self.extra_client_options
         return bigquery.Client(self.project_on_behalf, **client_options)
+    
+    def get_projects_client(self) -> resourcemanager_v3.ProjectsClient:
+        return resourcemanager_v3.ProjectsClient()
```

Comment:
Do you know if this client needs extra permission, or does it not need any new permission?
If it is needed, then can we make sure not to create this client if, in the config, it is not set to use it?

Reply:
Good question. I initially implemented this PR used the `list_projects` method but landed on the `search_projects` one due to its flexibility in searching for projects based on parents AND labels. Looking at [GCP docs](https://cloud.google.com/resource-manager/reference/rest/v3/projects/search), seems like you'll need to ensure the SA has `resourcemanager.projects.get` on the projects you want to grab.

Is that your read, too?

Reply:
@anaghshineh There is a easy way to test this out by creating a service account without this permission and try to run this code. If this does indeed require a new permission then please
- update the docs with the new roles/permissions required
- make this an optional thing disabled by default. We don't want every orgs ingestion to break due to this new permission. Some orgs have heavier checks on getting new permissions. We don't want them to get stuck on permissions when this gets in. 
- Please add in https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md that there is new option that requires new permissions

As this is open source we have to be careful w.r.t. not breaking other orgs ingestion as much as possible.